### PR TITLE
test_setup: refactor sriov_setup() and sriov_cleanup()

### PR DESCRIPTION
**Pull request by balamuruhans**
Removed redundant codes and made some cleanups to refactor sriov_setup()
and sriov_cleanup() methods

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/test_setup.py [hunk 1]
```diff
@@ -1333,6 +1334,54 @@ def is_binded_to_stub(self, full_id):
...
+        """
+        if not driver:
+            driver = self.driver
+        if driver and driver != "mlx5_core":
+            cmd = "modprobe -r %s" % driver
+        if ARCH == 'ppc64le' and driver == 'mlx5_core':
+            pf_devices = self.get_pf_mlx()
+            logging.info("Mellanox CX4 PF devices '%s'", pf_devices)
+            for PF in pf_devices:
+                self.set_vf(PF)
```
Comment (sathnaga) [inline virttest/test_setup.py]: use this return value to return this function

File: virttest/test_setup.py [hunk 2]
```diff
@@ -1333,6 +1334,54 @@ def is_binded_to_stub(self, full_id):
...
+        """
+        if not driver:
+            driver = self.driver
+        cmd = "modprobe %s" % driver
+        if option:
+            cmd += " %s" % option
+        msg = "Loading the driver '%s' without option" % driver
+        error_context.context(msg, logging.info)
+        return not bool(process.system(cmd, ignore_status=True, shell=True))
+
```
Comment (sathnaga) [inline virttest/test_setup.py]: setvf can be handled here

File: virttest/test_setup.py [hunk 3]
```diff
@@ -1389,75 +1435,62 @@ def sr_iov_setup(self):
...
             re_probe = True
+        # If driver is available then set VFs for ppc64le
         else:
-            self.setup = None
-            return True
+            if ARCH == 'ppc64le' and self.driver == 'mlx5_core':
+                pf_devices = self.get_pf_mlx()
+                for PF in pf_devices:
+                    if not self.set_vf(PF, self.driver_option):
+                        re_probe = True
```
Comment (sathnaga) [inline virttest/test_setup.py]: this can be placed inside probing funtions

**Comment by balamuruhans** [comment]
@sathnaga I have made changes as per your suggestion.

**Pull request merged**